KOGITO-3291: Implement security at Quarkus level for Management Console

nmirasch opened the pull request:
Related jira: https://issues.redhat.com/browse/KOGITO-3291

nmirasch commented:
jenkins retest this please

Sgitario commented on ui-packages/packages/common/src/utils/KeycloakClient.ts:
```diff
@@ -31,74 +35,143 @@ const keycloakInstance = createKeycloakInstance();
-        }
-      });
-
+    if (isReactAuthEnabled()) {
```
The name "isReactAuthEnabled" is accurated? Now, the auth is implemented in quarkus, so maybe just "isAuthEnabled" would be more correct?

Sgitario commented on ui-packages/packages/management-console/package.json:
```diff
@@ -12,7 +12,7 @@
     "build:prod": "yarn run lint && webpack --config webpack.prod.js",
     "start": "webpack-dev-server --hot --color --progress --info=true --config webpack.dev.js",
-    "start-auth": "webpack-dev-server --hot --color --progress --info=true --config webpack.dev.js --define process.env.KOGITO_AUTH_ENABLED=true",
+    "start-auth": "webpack-dev-server --hot --color --progress --info=true --config webpack.dev.js --define process.env.KOGITO_AUTH_ENABLED=true --define process.env.KOGITO_REACT_AUTH_ENABLED=true",
```
Why do we need this property "KOGITO_REACT_AUTH_ENABLED"? Should not we use the KOGITO_AUTH_ENABLED for both quarkus and react apps as they are tied with each other now?

nmirasch replied:
@Sgitario the idea is to maintain the two modes of security: with the quarkus application (oidc extension) and at the react side (keycloak.js). That reactAuth enabled (property "KOGITO_REACT_AUTH_ENABLED) has the goal to differentiate them.
The common use would be the quarkus security, but I keep this thinking in future integrations of the MC(react side).

Sgitario replied:
I was not aware of this functionality... Is this new property transparent of the overal usage of the KOGITO_AUTH_ENABLED? I mean if I enable the KOGITO_AUTH_ENABLED property, the security will be enabled without taking care of more properties?

nmirasch replied:
yes,  when security is enabled.. the property KOGITO_AUTH_ENABLED have to be true in both cases, and if its false is disabled. the only thing is, if the console want to be ran outside the quarkus app with security enabled, an extra property have to be set then :  KOGITO_REACT_AUTH_ENABLED.

Sgitario replied:
Ok, thanks for the explanation.

Sgitario reviewed (changes requested):
lgtm, but why do we need the KOGITO_REACT_AUTH_ENABLED new property?

Sgitario reviewed (approved):
lgtm

nmirasch commented:
jenkins retest this please

cristianonicolai commented on security-commons/.gitignore:
```diff
@@ -0,0 +1,14 @@
+/target
```
this file can probably removed, there is one at the root level already

cristianonicolai commented on security-commons/src/main/java/org/kie/kogito/security/UserResource.java:
```diff
@@ -0,0 +1,82 @@
+    public static class User {
+
+        private String userName = "Anonymous";
+        private Set<String> roles= Collections.emptySet();
```
```suggestion
        private Set<String> roles = Collections.emptySet();
```

cristianonicolai commented on security-commons/src/main/java/org/kie/kogito/security/UserResource.java:
```diff
@@ -0,0 +1,82 @@
+
+        User(SecurityIdentity identity) {
+            if(identity != null &&
+                    identity.getPrincipal()!=null &&
```
```suggestion
                    identity.getPrincipal() != null &&
```

cristianonicolai commented on security-commons/src/main/java/org/kie/kogito/security/UserResource.java:
```diff
@@ -0,0 +1,82 @@
+        User(SecurityIdentity identity) {
+            if(identity != null &&
+                    identity.getPrincipal()!=null &&
+                    identity.getCredential(TokenCredential.class)!=null) {
```
```suggestion
                    identity.getCredential(TokenCredential.class) != null) {
```

cristianonicolai reviewed (commented):
@nmirasch overall looks good to me, just missing two things I guess:
- Add the new security-commons module to kogito-bom just to be on the safe side for the managed version.
- Also, not sure if I missed but I would be good to have a IT test to check the endpoint when running the managent console app.

nmirasch commented:
depends on https://github.com/kiegroup/kogito-runtimes/pull/797

nmirasch commented:
@pefernan, @cristianonicolai @Sgitario  I have added the security-commons module to kogito-bom.. now this PR depends on  https://github.com/kiegroup/kogito-runtimes/pull/797. I have removed the security implementation at react part. Could you take a look please?

pefernan commented:
jenkins retest this please

pefernan reviewed (approved):
+1 with last changes looks and works great. Thanks!

cristianonicolai reviewed (approved):
great work @nmirasch !

pefernan commented:
jenkins retest this

jstastny-cz reviewed (approved):
LGTM
